packages/flutter/lib/src/gestures/selection_recognizers.dart

```diff
@@ -0,0 +1,514 @@
+        // resolve(GestureDisposition.rejected);
+        _checkTapUp(event);
+      }
+      _reset();
+    } else if (event is PointerCancelEvent) {
+      _checkLongPressCancel();
+      _reset();
+    } else if (event is PointerDownEvent) {
+      // The first touch.
+      _longPressOrigin = OffsetPair.fromEventPosition(event);
+      _initialButtons = event.buttons;
+      _checkTapDown(event);
+    } else if (event is PointerMoveEvent) {
+      if (event.buttons != _initialButtons) {
+        resolve(GestureDisposition.rejected);
+        stopTrackingPointer(primaryPointer!);
+      } else if (_longPressAccepted) {
+        _checkLongPressMoveUpdate(event);
+      }
+    }
+  }
+
+  void _checkTapDown(PointerDownEvent event) {
+    assert(_longPressOrigin != null);
+    print('from recognizer check tap down');
+    final TapDownDetails details = TapDownDetails(
+      globalPosition: _longPressOrigin!.global,
+      localPosition: _longPressOrigin!.local,
+      kind: getKindForPointer(event.pointer),
+    );
+
+    if (lastTapOffset == null) {
+      consecutiveTapCount += 1;
+      lastTapOffset = details.globalPosition;
+    } else {
+      if (consecutiveTapTimer != null && isWithinConsecutiveTapTolerance(details.globalPosition)) {
+        consecutiveTapCount += 1;
+        consecutiveTapTimerReset();
+      }
+    }
+
+    _isDoubleTap = consecutiveTapCount == 2;
+
+    switch (_initialButtons) {
+      case kPrimaryButton:
+        if (onTapDown != null) {
+          invokeCallback<void>('onTapDown', () => onTapDown!(details, consecutiveTapCount));
+        }
+        break;
+      default:
+        assert(false, 'Unhandled button $_initialButtons');
+    }
+  }
+
+  void _checkTapUp(PointerUpEvent event) {
+    print('from recognizer check tap up');
+    final TapUpDetails details = TapUpDetails(
+      globalPosition: event.position,
+      localPosition: event.localPosition,
+      kind: getKindForPointer(event.pointer),
+    );
+
+    _velocityTracker = null;
+    switch (_initialButtons) {
+      case kPrimaryButton:
+        if (onTapUp != null) {
+          invokeCallback<void>('onTapUp', () => onTapUp!(details, consecutiveTapCount));
+        }
+        break;
+      default:
+        assert(false, 'Unhandled button $_initialButtons');
+    }
+    consecutiveTapTimer ??= Timer(kDoubleTapTimeout, consecutiveTapTimeout);
+  }
+
+  void _checkLongPressCancel() {
+    print('from recognizer check tap cancel');
+    if (state == GestureRecognizerState.possible) {
+      switch (_initialButtons) {
+        case kPrimaryButton:
+          if (onLongPressCancel != null) {
+            invokeCallback<void>('onLongPressCancel', onLongPressCancel!);
+          }
+          break;
+        default:
+          assert(false, 'Unhandled button $_initialButtons');
+      }
+    }
+  }
+
+  void _checkLongPressStart() {
+    print('from recognizer check long press start');
+    if (_isDoubleTap) {
```

Comment:
does this work for double tap and drag?

Reply:
Do you mean if we can do this case for double tap and drag?, or if this case breaks double tap to drag?